tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,130 @@
+use transactional qw(process_reboot trup_install trup_shell);
+use File::Basename;
+use version_utils;
+use Data::Dumper;
+sub install_dependency_package {
+    my ($instance, $rke2_url, $kubectl_url, $helm_url) = @_;
+    trup_call("pkg install curl git docker");
+    process_reboot(trigger => 1);
+    systemctl("enable docker");
+    systemctl("start docker");
+    systemctl("status docker");
+    script_run("curl -sSL $rke2_url -o ./install_rke2.sh && chmod 775 ./install_rke2.sh");
+    script_run("sh ./install_rke2.sh");
+    script_run("echo 'export PATH=\$PATH:/opt/rke2/bin' >> ~/.bashrc");
+    systemctl("enable rke2-server.service");
+    systemctl("start rke2-server.service");
+    systemctl("status rke2-server.service");
+    script_run("rke2 --version");
+    script_run("curl -sSL $helm_url -o ./install_helm.sh && chmod 775 ./install_helm.sh");
+    script_run("sh ./install_helm.sh");
+    script_run("helm version");
+    script_run("curl -sSL $kubectl_url -o ./kubectl && chmod +x ./kubectl");
+    script_run("sudo mv ./kubectl /usr/local/bin/");
+    script_run("kubectl version --client");
+}
+
+sub install_aistack_chart {
+    my ($instance, $cert_repo, $ing_repo) = @_;
+    record_info('Install AISTACK helm charts');
+    my $username = $ENV{'USER'};
+    my $curl_options = "-sSL --retry 3 --retry-delay 30";
+    my $APPLICATION_COLLECTION_USER_PASSWORD_SECRET ="mysecretpassword";
+    my $docker_user_name = 'yarunachalam@suse.com';
+
+    set_var('APPLICATION_COLLECTION_USER_PASSWORD_SECRET', '<application_collection_user_token>');
+
+
+
+    # config kubectl
+    assert_script_run("mkdir -p ~/.kube");
+    assert_script_run("sudo cp /etc/rancher/rke2/rke2.yaml ~/.kube/config");
+    # assert_script_run("sudo chown $username:$username ~/.kube/config");
+    assert_script_run("kubectl config get-contexts");
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config view");
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manger --create-namespace --version v1.15.2 --set crds.enabled=true");
+
+    sleep 180;
+    assert_script_run("helm list --all-namespaces");
+    assert_script_run("kubectl get pods --all-namespaces");
+
+    # Access to Application collection registery
+    # Get docker username and password
+    assert_script_run("kubectl create ns suse-private-ai");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$APPLICATION_COLLECTION_USER_PASSWORD_SECRET' -n suse-private-ai", timeout => 360);
+    # Add suse-private-ai install
+    my $helm_values = get_var('HELM_CONFIG', 'https://github.com/yarunachalam/os-autoinst-distri-opensuse/tree/aistack_basic/data/aistack_values.yaml');
```

Comment:
When this PR will be merged, the `aistack_values.yaml` shall be retrieved from `https://github.com/os-autoinst/....`: instead here you set as **default** your own forked repo `https://github.com/yarunachalam/...` version. 
It can be a temporary draft, to be changed before code ready.

Reply:
I agree

Reply:
Code changed, closing this outdated post.